src/EVM/ABI.hs

```diff
@@ -525,6 +526,27 @@ decodeBuf tps buf =
   where
     isDynamic t = abiKind t == Dynamic
 
+decudeBufRobust :: [AbiType] -> Expr Buf -> (AbiVals, String)
```

Comment:
Can we fix the existing method instead of introducing new one?
We can adjust the types to better convey the necessary information.

`decodeBuff` can return an `Either`, where `Left` can pass along the error string. `Right` can be `AbiVals` where we remove `NoVals`.
We can extend it with `Maybe ByteString` to return the unprocessed bytes back, if there are any.

I would prefer to leave it up to the caller to decide what to do in such case, and not decide on the message here in the callee.

I was thinking that in our use case, where we want to pretty print the Solidity function call, the extra bytes do not pose any problem and we can safely ignore them, no?

Reply:
`the extra bytes do not pose any problem` -> in this case, they don't seem to. But we also use this method to decode things for cheatcodes, for example, where we cannot ignore it. Either would work there, actually. Let's say that I don't want to warn the user about the trailing bytes (not my preference, but OK). I'd still need to return a Left, because some functions rely on this being exact. Should I return a `Left AbiVals`?

Reply:
What do you mean by "relying on this being exact"?
My idea is that the return type would be something like `Either String (AbiVals, Maybe ByteString)`.
The callers then implement their way of handling each situation.
Which could be an error, or success with all bytes consumed, or success with some leftover bytes.